qutebrowser/browser/hints.py

```diff
@@ -720,6 +720,7 @@ def _handle_auto_follow(self, keystr="", filterstr="", visible=None):
                        if label.isVisible()}
 
         if len(visible) != 1:
+            self._apply_auto_follow_timeout()
```

Comment:
Actually, this is not the right place for the number mode, since `filter_hints` can [return before calling _handle_auto_follow](https://github.com/sfgasdfsafggfgg/qutebrowser/blob/fc1af906a9e8727a8b0a245a86c85dadcb3262ed/qutebrowser/browser/hints.py#L806) if there are no visible hints.